base/utf16.jl

```diff
@@ -66,15 +76,190 @@ function encode16(s::AbstractString)
     UTF16String(buf)
 end
 
+function ascii_to_utf16(s::Array{UInt8,1})
+    len::Int = sizeof(s)
+    buf = Array{UInt16,1}(len+1)
+    for i = 1:len
+        buf[i] = s[i]
+    end
+    buf[len+1] = 0 # NULL termination
+    UTF16String(buf)
+end
+
+err_utf8surr(i::Integer, c::Unsigned) = ArgumentError("invalid UTF-8 byte at index $i (0x$(hex(c)) is not leading byte for a surrogate character)")
+err_utf8invalid(i::Integer, c::Unsigned) = ArgumentError("invalid UTF-8 byte at index $i (0x$(hex(c)))")
+err_utf8short(i::Integer, c::Unsigned) = ArgumentError("invalid UTF-8 sequence starting at index $i (0x$(hex(c)) missing one or more continuation bytes)")
+err_utf8cont(i::Integer, c::Unsigned) = ArgumentError("invalid UTF-8 sequence starting at index $i (0x$(hex(c)) is not a continuation byte)")
+err_utf8long(i::Integer, c::Unsigned) = ArgumentError("invalid UTF-8 overlong encoding starting at index $i (0x$(hex(c)))")
+err_missingsurr(i::Integer, c::Unsigned) = ArgumentError("missing trailing Unicode surrogate character after index $i (0x$(hex(c))")
+err_notleadsurr(i::Integer, c::Unsigned) = ArgumentError("not a leading Unicode surrogate character at index $i (0x$(hex(c)))")
+err_nottrailsurr(i::Integer, c::Unsigned) = ArgumentError("not a trailing Unicode surrogate character at index $i (0x$(hex(c)))")
+err_invalidchar(i::Integer, c::Unsigned) = ArgumentError("invalid Unicode character starting at index $i (0x$(hex(c)) > 0x10ffff)")
+
+function utf8_to_utf16(s::Array{UInt8,1})
+    len::Int = sizeof(s)
+    # handle zero length string quickly
+    if len == 0
+        buf = Array{UInt16,1}(1)
+        buf[1] = 0
+        return UTF16String(buf)
```

Comment:
Ah, good... I didn't know you could just say UInt16[0] to get an array with just the 0. I'll change those.